### ISSUE TITLE
Using dotnetfiddle.net / a no-setup beginner approach?

### ISSUE THREAD

**Issue by ylluminate** [Almantask/CSharp-From-Zero-To-Hero#487]
I recently ran into a situation with some learners that are having a hard time with C# primarily due to setup. Because of their desire to learn Unity / game development the exposure to C# has been rough and they don't really have the patience to move through the initial setup of straight C# development.  

It would be really great if some intermediate / beginner option that isolated the development environment setup from the language learning were available. I think that learning to set up the dev env after getting their hands around the language for a while first would yield better results.

**Comment by Almantask** [comment]
That's a good idea. Is there any guide on how to integrate automated tests with .NET Fiddle?

**Comment by Matt500b** [comment]
Definitely like the appeal of this - quickly whipping up a repl.it or something for the first few homeworks would be neat for the true beginner beginners, and then move onto an IDE once they're a bit more confident. Only issues is they wont be using GitHub and get the benefit of the build, testing etc. But it could be a way to ease someone in.

Had a look around..
- [.NET Fiddle](https://dotnetfiddle.net/) doesn't have integrated testing
- [repl.it](https://repl.it/site/classrooms) doesn't have it by default, it does have a teacher section but limited to 200 students for the free plan. [pricing](https://repl.it/site/classrooms), [auto-grading](https://docs.repl.it/classrooms/autograding)
- [jdoodle](https://www.jdoodle.com/) has a [guru](https://docs.jdoodle.com/guru/assignments) mode which seems to be completely free - however the sandbox for coding is no where near as good as the others and the unit testing seems a bit flakey, not really testing it out

There are probably others but the few i've looked at.

**Comment by ylluminate** [comment]
Yes, I've been grappling with these thoughts as well.  I've come to the conclusion that using `git` is probably best when pushed out by a goodly amount of experience first. Version control has been a hinderance each time I've tried to teach it or share tutorials with students. Even after they've "learned" it, I find that they refuse to use it until they get bitten really badly by a mistake or some accident and wished they'd have some history data.

I have even tried to push Godot first as has been suggested by some educators due to the lower GDscript learning curve and its seeming "subset of C#" conceptuality... but then I've had students push back saying, "no thanks, I want to do Unity with C#".

I believe the present educational paradigm and huge gaming ecosystem is a double edged sword that is very defeative for personality types that would have benefited from less structured learning and more educational exploration and self direction... and unfortunately this "no-setup beginner approach" is necessary given this current social climate.

**Comment by deccer** [comment]
There is also [LinqPad](https://www.linqpad.net) and [RoslynPad](https://github.com/aelij/roslynpad)
